### PR TITLE
Misc code clean up, 2nd part

### PR DESCRIPTION
**Pull request by vks**
This pull request contains a lot of changes modernizing the code, in the same spirit as #1284. All of them are hopefully uncontroversial.

Two serious use-after-free bugs found be clippy were fixed.

### CODE REVIEW

File: src/unlink/unlink.rs [hunk 1]
```diff
@@ -68,7 +68,8 @@ pub fn uumain(args: Vec<String>) -> i32 {
         );
     }
 
-    let c_string = CString::new(matches.free[0].clone()).unwrap(); // unwrap() cannot fail, the string comes from argv so it cannot contain a \0.
+    let c_string = CString::new(matches.free[0].clone()).unwrap();
+    //^ unwrap() cannot fail, the string comes from argv so it cannot contain a \0.
```
Comment (Arcterus) [inline src/unlink/unlink.rs]: Move this to the line before `let c_string = ...`.

File: src/mv/mv.rs [hunk 2]
```diff
@@ -202,12 +202,12 @@ fn determine_backup_suffix(backup_mode: BackupMode, matches: &getopts::Matches)
                 );
             }
         }
+    } else if let (Ok(s), BackupMode::SimpleBackup) =
+        (env::var("SIMPLE_BACKUP_SUFFIX"), backup_mode)
```
[the remark this answers is not in the document]
Comment (vks) [reply src/mv/mv.rs]: How would you write it instead?

File: src/mv/mv.rs [hunk 3]
```diff
@@ -349,24 +349,23 @@ fn rename(from: &PathBuf, to: &PathBuf, b: &Behaviour) -> Result<()> {
...
+            fs::rename(to, p)?;
         }
 
-        if b.update {
-            if try!(try!(fs::metadata(from)).modified()) <= try!(try!(fs::metadata(to)).modified())
-            {
-                return Ok(());
-            }
+        if b.update &&
+           (fs::metadata(from))?.modified()? <= fs::metadata(to)?.modified()?
```
[the remark this answers is not in the document]
Comment (vks) [reply src/mv/mv.rs]: I believe this violates the rustfmt convention, but I don't really care.

File: src/nl/nl.rs [hunk 4]
```diff
@@ -333,10 +333,11 @@ fn nl<T: Read>(reader: &mut BufReader<T>, settings: &Settings) {
         // way, start counting empties from zero once more.
         empty_line_count = 0;
         // A line number is to be printed.
-        let mut w: usize = 0;
-        if settings.number_width > line_no_width {
-            w = settings.number_width - line_no_width;
-        }
+        let w: usize = if settings.number_width > line_no_width {
```
[the remark this answers is not in the document]
Comment (vks) [reply src/nl/nl.rs]: No, I removed it.

File: src/unlink/unlink.rs (hunk 1 above)
Comment (vks) [reply src/unlink/unlink.rs]: Done.

File: src/mkfifo/mkfifo.rs [hunk 5]
```diff
@@ -75,11 +75,12 @@ Create a FIFO with the given name.",
 
     let mut exit_status = 0;
     for f in &matches.free {
-        let err = unsafe {
-            mkfifo(
-                CString::new(f.as_bytes()).unwrap().as_ptr(),
+        let err = {
+            let path = CString::new(f.as_bytes()).unwrap();
+            unsafe { mkfifo(
```
[the remark this answers is not in the document]
Comment (vks) [reply src/mkfifo/mkfifo.rs]: Done.

**Comment by sylvestre** [comment]
I am sorry about this but I will close it. @rivy did a bunch of similar changes lately.

**Pull request closed without merging**